src/test/java/uk/gov/pay/ledger/queue/EventDigestHandlerTest.java

```diff
@@ -0,0 +1,116 @@
+
+@RunWith(MockitoJUnitRunner.class)
+public class EventDigestHandlerTest {
+
+    @Mock
+    private EventService eventService;
+    @Mock
+    private TransactionService transactionService;
+    @Mock
+    private TransactionMetadataService transactionMetadataService;
+    @Mock
+    private PayoutService payoutService;
+    @Captor
+    private ArgumentCaptor<LoggingEvent> loggingEventArgumentCaptor;
+    @Mock
+    private Appender<ILoggingEvent> mockAppender;
+
+    @InjectMocks
+    private EventDigestHandler eventDigestHandler;
+    private EventDigest eventDigest;
+
+    @Before
+    public void setUp() {
+        eventDigest = EventDigest.fromEventList(List.of(anEventFixture().toEntity()));
+        when(eventService.getEventDigestForResource(any()))
+                .thenReturn(eventDigest);
+    }
+
+    @Test
+    public void shouldUpsertTransactionIfResourceTypeIsPayment() {
+        Event event = anEventFixture().withResourceType(PAYMENT).toEntity();
+        eventDigestHandler.processEvent(event);
+
+        verify(eventService).getEventDigestForResource(event.getResourceExternalId());
+        verify(transactionService).upsertTransactionFor(eventDigest);
+        verify(transactionMetadataService).upsertMetadataFor(event);
+    }
+
+    @Test
+    public void shouldUpsertTransactionIfResourceTypeIsRefund() {
+        Event event = anEventFixture().withResourceType(REFUND).toEntity();
+        eventDigestHandler.processEvent(event);
+
+        verify(eventService).getEventDigestForResource(event.getResourceExternalId());
+        verify(transactionService).upsertTransactionFor(eventDigest);
+        verify(transactionMetadataService).upsertMetadataFor(event);
+    }
+
+    @Test
+    public void shouldUpsertPayoutIfResourceTypeIsPayout() {
+        Event event = anEventFixture().withResourceType(PAYOUT).toEntity();
+        eventDigestHandler.processEvent(event);
+
+        verify(eventService).getEventDigestForResource(event.getResourceExternalId());
+        verify(payoutService).upsertPayoutFor(eventDigest);
+    }
+
+    @Test
+    public void shouldLogWarningIfResourceTypeIsNotPayout() {
+        Logger root = (Logger) LoggerFactory.getLogger(EventDigestHandler.class);
+        root.addAppender(mockAppender);
+
+        Event event = anEventFixture().withResourceType(AGREEMENT).toEntity();
+        eventDigestHandler.processEvent(event);
+
+        verify(transactionService, never()).upsertTransactionFor(any());
+        verify(transactionMetadataService, never()).upsertMetadataFor(any());
+        verify(payoutService, never()).upsertPayoutFor(any());
+
+        verify(mockAppender, times(1)).doAppend(loggingEventArgumentCaptor.capture());
+        List<LoggingEvent> loggingEvents = loggingEventArgumentCaptor.getAllValues();
+        assertThat(loggingEvents.get(0).getFormattedMessage(),
+                is(format("Event digest processing for resource type [%s] is not supported. " +
```

Comment:
Potentially don't need to assert on the exact string logged as that may make it brittle in the future? Maybe a substring match like `resource type is not supported` or something would do. Nothing that needs to change now but can just see this coming up